sdk/src/main/kotlin/io/rover/sdk/ui/blocks/poll/OptionResults.kt

```diff
@@ -18,4 +18,36 @@ internal data class OptionResults(val results: Map<String, Int>) {
             )
         }
     }
+}
+
+private data class VotesWithFractional(val key: String, var votePercentage: Int, val fractional: Float)
+
+internal fun changeVotesToPercentages(results: OptionResults): OptionResults {
+    // https://en.wikipedia.org/wiki/Largest_remainder_method
+    val total = results.results.values.sum().toFloat()
+    val votes = results.results.mapValues { (it.value.toFloat() / total * 100) }
+
+    val votesWithFractional = votes.toList().map {
+        VotesWithFractional(
+            it.first,
+            it.second.toInt(),
+            it.second - it.second.toInt()
+        )
+    }
+
+    var differenceBetweenVotesAndTotalPercentage = 100 - votesWithFractional.sumBy { it.votePercentage }
+
+    val votesList = votesWithFractional.sortedByDescending { it.fractional }.toMutableList()
```

Comment:
I think you might have the same issue here I did on iOS.  The sort order of votesWithFractional is undefined before you sort by the fractional value.  However, if any of them have the same `fractional` value, then you need something to disambiguate the order.  What I did was pre-sort the options by the option keys.

Reply:
Because it's a stable sort, I think that the order will alway be the same in ties if the results are in the same order

Reply:
The results should be in the same map entry order every time so the `votesWithFractional` will have an ordering and `sortedByDescending` is a stable sort, so that should preserve that ordering on ties